src/controllers/user.js

```diff
@@ -39,19 +36,114 @@ class UserController {
 
       const secret = process.env.TOKEN_SECRET;
 
-      const duration = {
-        expiresIn: process.env.TOKEN_EXPIRE,
-      };
+      const duration = process.env.TOKEN_EXPIRE;
 
       const token = generateToken(params, secret, duration);
 
-      res.header('Authorization', token);
+      // Send a confirmation email
+      const email = await this.sendConfirmationEmail(user.email);
+
+      res.header('Authorization', `Bearer ${token}`);
       res.status(200).json({
         message: 'User created successfully',
         token,
+        email,
       });
     } catch (error) {
-      res.status(400).json({ error: error.message });
+      res.status(400).json({ error: 'email already exists' });
```

Comment:
This is a wrong assumption, any other errors can occur, eg a service not being available. eg db service